fgpyo/sam/__init__.py

```diff
@@ -606,6 +607,110 @@ def __getitem__(
         return self.elements[index]
 
 
+@enum.unique
+class PairOrientation(enum.Enum):
+    """Enumerations of read pair orientations."""
+
+    FR = "FR"
+    """A pair orientation for forward-reverse reads ("innie")."""
+
+    RF = "RF"
+    """A pair orientation for reverse-forward reads ("outie")."""
+
+    TANDEM = "TANDEM"
+    """A pair orientation for tandem (forward-forward or reverse-reverse) reads."""
+
+    @classmethod
+    def build(
```

Comment:
How would you feel about calling this `from()` or similar?  It feels wrong to me to have a `build()` on an enum.

Reply:
`from` is a reserved keyword in Python so that wouldn't work syntactically. I saw a `build()` on `Template` so copied that word:

https://github.com/fulcrumgenomics/fgpyo/blob/3680b4fe1fe3d8686172e05dbf9c5043fa51ae1a/fgpyo/sam/__init__.py#L858

I could do:

- `of(rec1, rec2)`
- `from_recs(rec1, rec2)`

Do you like any of those better?

Reply:
Either one works for me.  I just think `build()` is strange since it's _selecting_ an enum value, not building a new one.

Reply:
I went with `from_recs(rec1, rec2)` :)

Reply:
I tend to go with `.from_FOO()` for these sorts of things, FWIW